### PR TITLE
Visitor can see products sorted in different categories

### PR DESCRIPTION
**Pull request by piavW**
[PT-link](https://www.pivotaltracker.com/story/show/168829430)
By: Pia & Yasmine
## PR content
In this PR we've sorted the restaurants products into three different categories: Starters, Main Courses, Desserts.

## This PR contains
- Category model
- Unit testing for model
- Feature testing for view of categories
- Association between different models (categories has many products)

## What we've learned
- How to do association with Rails

Screenshot 2019-10-02 at 17 05 49

### CODE REVIEW

File: features/category.feature [hunk 1]
```diff
@@ -0,0 +1,29 @@
+Feature: Visitor can view menu products ordered in categories
+ As a visitor,
```
Comment (clalil) [inline features/category.feature]: Indentation please. :)

File: features/step_definitions/basic_steps.rb [hunk 2]
```diff
@@ -2,22 +2,37 @@
   visit root_path
 end
 
-Given("I click {string}") do |element|
+Given("I click {string}")do |element|
```
Comment (clalil) [inline features/step_definitions/basic_steps.rb]: Space please!

File: features/step_definitions/basic_steps.rb [hunk 3]
```diff
@@ -2,22 +2,37 @@
...
   click_on element
 end
 
 When("I fill in {string} with {string}") do |field, content|
-  fill_in field, with: content
+    fill_in field, with: content
```
Comment (clalil) [inline features/step_definitions/basic_steps.rb]: Indentation.

File: features/step_definitions/basic_steps.rb [hunk 4]
```diff
@@ -2,22 +2,37 @@
...
 end
 
 Given("the following user exists") do |table|
   table.hashes.each do |user|
     FactoryBot.create(:user, user)
   end
-end
\ No newline at end of file
+end
+# Given("The following cars exist") do |table|
```
Comment (clalil) [inline features/step_definitions/basic_steps.rb]: Remove, commented lines, please.

File: config/routes.rb [hunk 5]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  resources :category
```
Comment (oliverochman) [inline config/routes.rb]: do you need the category routes?

File: features/category.feature [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  In order to choose product more easily
+  I would like to see the products sorted in categorie
+
+  Background: 
+    Given the following categories exists
+      | title         |
+      | Starters      |
+      | Main Courses  |
+      | Desserts      |
+    Given the following products exists
```
Comment (oliverochman) [inline features/category.feature]: `And the following products exists`

File: features/category.feature [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+      | name          | description                     | price | category        |
+      | garlic bread  | garlic, bread, butter           | 30    | Starters        |
+      | margherita    | cheese, tomato sauce, basil     | 70    | Main Courses    |
+      | vesuvio       | cheese, tomato sauce, ham       | 70    | Main Courses    |
+      | calzone       | cheese, tomato sauce, ham       | 80    | Main Courses    |
+      | ice-cream     | strawberry                      | 40    | Desserts        |
+      | ice-cream     | vanilla                         | 40    | Desserts        |
+    Given I visit the landing page
```
Comment (oliverochman) [inline features/category.feature]: `And I visit the landing page`

File: features/category.feature [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    Then I should see 'Menu'
```
Comment (oliverochman) [inline features/category.feature]: Unnecessary assertion, please remove it

File: spec/models/category_spec.rb [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+  
+  describe 'DB table' do
+    it { is_expected.to have_db_column :id }
+    it { is_expected.to have_db_column :title }
+  end
+
+  describe 'Validations' do
+    it { is_expected.to validate_presence_of :title }
+  end
+
```
Comment (oliverochman) [inline spec/models/category_spec.rb:17]: you need to test the association between category and product in both model specs

File: app/controllers/products_controller.rb [hunk 10]
```diff
@@ -1,5 +1,6 @@
 class ProductsController < ApplicationController
   def index
     @products = Product.all
```
Comment (oliverochman) [inline app/controllers/products_controller.rb]: do you need this variable now?

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Review by clalil — APPROVED** [review]

**Review by yasmineezequiel — APPROVED** [review]
Awesome!

File: app/controllers/products_controller.rb [hunk 11]
```diff
@@ -1,5 +1,6 @@
 class ProductsController < ApplicationController
   def index
-    @products = Product.all
+    # @products = Product.all
```
Comment (oliverochman) [inline app/controllers/products_controller.rb]: please remove commented out code

File: config/routes.rb [hunk 12]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  # resources :category
```
Comment (oliverochman) [inline config/routes.rb]: please remove commented out code

File: spec/models/category_spec.rb [hunk 13]
```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe Category, type: :model do
+  
+  describe 'DB table' do
+    it { is_expected.to have_db_column :id }
+    it { is_expected.to have_db_column :title }
+    it { should have_many( :products) }
```
Comment (oliverochman) [inline spec/models/category_spec.rb]: please put this in its own description block

```
 describe 'Associations' do
    it { should have_many( :products) }
end

File: spec/models/product_spec.rb [hunk 14]
```diff
@@ -7,6 +7,7 @@
     it { is_expected.to have_db_column :name }
     it { is_expected.to have_db_column :description }
     it { is_expected.to have_db_column :price }
+    it { should belong_to( :category) }
```
Comment (oliverochman) [inline spec/models/product_spec.rb]: same here

**Review by oliverochman — CHANGES_REQUESTED** [review]

**Review by oliverochman — APPROVED** [review]

**Pull request merged**